### DISCUSSION TITLE
Does getRunningQueriesThunk work with clients as well?

### DISCUSSION THREAD

**Discussion by dev-splin** [reduxjs/redux-toolkit#3891] [Q&A]
I am calling multiple APIs using the RTK Query focus function.

I want to move the page to router.push at the same time as focus after the API called focus is over.

So I tried getRunningQueriesThunk, but it didn't work the way I wanted it to.

Can't wait for RTK Query running on getRunningQueriesThunk from client?

**Comment by phryneas** [comment]
I'm sorry, I'm not 100% sure what you're asking. Could you please add some more detail to the question (what do you mean by client) and show some code?

**Reply by dev-splin** [reply]
I mean as below.
When the button is pressed, there is logic to run router.push and API to call when focus.
If you are on another browser tab and immediately press the button, the API runs during focus, and router.push runs.
At this time, I want to run router.push after the API runs during focus.

What should I do?


```javascript
// in component

const { data } = useFetchInfo(1, { refetchOnFocus: true }); // API running during focus

...

return <Button onClick={async () => {
  await Promise.all(dispatch(getRunningQueriesThunk())); // Logic to run after API in progress
  router.push('/home');'
}}/>;
```

**Reply by phryneas** [reply]
It will work within the same app, but not between browser tabs.
